blockchain/txvalidator.go

```diff
@@ -1603,21 +1603,29 @@ func (b *BlockChain) checkCRCProposalTrackingTransaction(txn *Transaction,
 		return errors.New("invalid payload")
 	}
 
-	// Check if the proposal exist.
+	// Check if proposal exist.
 	proposalState := b.crCommittee.GetProposalManager().GetProposal(
 		cptPayload.ProposalHash)
 	if proposalState == nil {
 		return errors.New("proposal not exist")
 	}
 
+	// Check proposal status.
 	if proposalState.Status != crstate.VoterAgreed {
 		return errors.New("proposal status is not VoterAgreed")
 	}
 
+	// Check proposal tracking count.
 	if proposalState.TrackingCount >= b.chainParams.MaxProposalTrackingCount {
 		return errors.New("reached max tracking count")
 	}
 
+	// Check draft data of proposal tracking document.
+	dataHash := common.Uint256(common.Sha256D(cptPayload.DocumentData))
```

Comment:
maybe add  a DocumentHash() function to payload is more proper

Reply:
added